doc(STONEINTG-1052): cancelling an integration test (#307)
konflux-ci/docs: modules/testing/pages/integration/canceling.adoc

= Canceling Integration pipelineruns for Pull Requests

Pushing a new change to your Pull Request creates a new build pipelineRun, followed by a new snapshot if build succeeded which then triggers integration test pipelineRuns for that snapshot.
In case you happen to push another change in your Pull Request while integration tests pipelineRuns are still running, the previous test pipelineRuns will be canceled.

== How do I cancel the previous pipelinerun and which will be canceled?

If your integration tests are still running and you push new changes to your Pull Request, once new build pipelinerun succeeds and new snapshot is created,

> integration service will gracefully cancel previous integration test pipelinerun.

Reviewer: I feel like this is just restating the previous paragraph. Can we combine them into one concise statement? If we just have one paragraph, does it make more sense to integrate this content into another page?

Author: Combined those two paragraphs into one. I think the logic was preserved.

Reviewer: I think that this page is still confusing. When I see this topic, I feel like it will describe to me how _I_ can cancel ITS runs. But I think this is just stating how ITS self-cancels outdated PLRs? 

If so, can we combine this with a different page?

Reviewer 2: Would it help if we expanded the document to also describe how to manually cancel an ongoing ITS run from the Konflux UI?
It would be basically mentioning that you can press the `cancel` option from the dropdown menu next to the pipelineRun in the UI:
![image](https://github.com/user-attachments/assets/39c1a695-a61d-4969-b725-fe13d1e933a0)

Reviewer: Yeah, I think that would make it better.

= Canceling Integration pipelineruns for Pull Requests

> Pushing a new change to your Pull Request creates a new build pipelinerun, followed by a new snapshot if build succeeded, then integration tests are triggered.

Reviewer 2: ```suggestion
Pushing a new change to your Pull Request creates a new build pipelineRun, followed by a new snapshot if build succeeded which then triggers integration test pipelineRuns for that snapshot.
```

> In case you happen to push another change in your Pull Request, while integration tests pipelinerun are still running, the previous test pipelinerun will be canceled.

Reviewer 2: ```suggestion
In case you happen to push another change in your Pull Request while integration tests pipelineruns are still running, the previous test pipelineruns will be canceled.
```

== How do I cancel the previous pipelinerun and which will be canceled?

If your integration tests are still running and you push new changes to your Pull Request, once new build pipelinerun succeeds and new snapshot is created,
integration service will gracefully cancel previous integration test pipelinerun.
Each time newer snapshot is created it will be registered by integration service as a latest one, everything after that is being considered obsolete. That means

> old snapshots are being marked as canceled, same applies for integration test pipelineruns.

Reviewer 2: ```suggestion
old snapshots will be marked as canceled, same applies for integration test pipelineRuns.
```

= Canceling Integration pipelineruns for Pull Requests

> Pushing a new change to your PR creates new build pipelinerun, followed by a new snapshot if build succeeded, then integration tests are triggered.

Reviewer 3: ```suggestion
Pushing a new change to your PR creates a new build pipelinerun, followed by a new snapshot if build succeeded, then integration tests are triggered.
```

> Pushing a new change to your PR creates a new build pipelinerun, followed by a new snapshot if build succeeded, then integration tests are triggered.

Reviewer 4: ```suggestion
Pushing a new change to your Pull Request creates a new build pipelinerun, followed by a new snapshot if build succeeded, then integration tests are triggered.
```

> In case you happen to push another change in your Pull Request, while integration tests pipelinerun are still running, the previous pipelinerun will be canceled.

Reviewer 4: ```suggestion
In case you happen to push another change in your Pull Request, while integration tests pipelinerun are still running, the previous test pipelinerun will be canceled.
```

== How do I cancel the previous pipelinerun?

> If your integration tests are still running and you push new changes to your Pull Request, once build pipelinerun finishes with success and new snapshot is created,

Reviewer 4: ```suggestion
If your integration tests are still running and you push new changes to your Pull Request, once new build pipelinerun succeeds and new snapshot is created,
```

> this change will be detected by integration service and will gracefully cancel previous integration test pipelinerun.

Reviewer 4: ```suggestion
integration service will gracefully cancel previous integration test pipelinerun.
```

> == Which integration pipelinerun will be canceled?

Reviewer 4: I feel this part can be combined with the last one since it has mentioned when older integration test pipelinerun is canceled.

Author: combined those two together
